src/simplnx/Utilities/Math/GeometryMath.hpp

```diff
@@ -23,6 +23,49 @@ class VertexGeom;
 
 namespace GeometryMath
 {
+namespace detail
+{
+struct GeometryStoreCache
+{
+  GeometryStoreCache(const AbstractDataStore<float32>& verticesStore, const AbstractDataStore<IGeometry::MeshIndexType>& facesStore, usize numVertsPerFace)
+  : VerticesStoreRef(verticesStore)
+  , FacesStoreRef(facesStore)
+  , NumVertsPerFace(numVertsPerFace)
+  {
+  }
+
+  const AbstractDataStore<float32>& VerticesStoreRef;
+  const AbstractDataStore<IGeometry::MeshIndexType>& FacesStoreRef;
+  usize NumVertsPerFace;
+};
+
+template <typename T>
+concept FloatType = std::is_floating_point_v<T>;
```

Comment:
The standard actually has this one: https://en.cppreference.com/w/cpp/concepts/floating_point